### PR TITLE
removing findDOMNode

### PR DESCRIPTION
**Pull request by zanderz**
This leaves just one instance of findDOMnode, in FloatingBox, which seems like it could be quite complicated to change & test. Making a separate ticket for that.

Also, Danny has a change pending to make the emoji focus work again.

### CODE REVIEW

File: shared/chat/conversation/messages/wrapper/exploding-height-retainer/index.desktop.js [hunk 1]
```diff
@@ -24,9 +23,17 @@ type State = {
   height: number,
 }
 class ExplodingHeightRetainer extends React.PureComponent<Props, State> {
+  boxRef: React.ElementRef<any>
```
Comment (chrisnojima) [inline shared/chat/conversation/messages/wrapper/exploding-height-retainer/index.desktop.js]: these should start w/ underscore, why use any

File: shared/chat/conversation/messages/wrapper/exploding-height-retainer/index.desktop.js [hunk 2]
```diff
@@ -24,9 +23,17 @@ type State = {
...
+  setHeight: () => void
   state = {animating: false, children: copyChildren(this.props.children), height: 17}
   timerID: SharedTimerID
 
+  constructor(props: Props) {
+    super(props)
+    // create a ref to store the textInput DOM element
+    this.boxRef = React.createRef()
```
Comment (chrisnojima) [inline shared/chat/conversation/messages/wrapper/exploding-height-retainer/index.desktop.js]: this can just be initialized above

File: shared/chat/conversation/messages/wrapper/exploding-height-retainer/index.desktop.js [hunk 3]
```diff
@@ -23,15 +23,15 @@ type State = {
...
+  _boxRef = React.createRef()
   setHeight: () => void
   state = {animating: false, children: copyChildren(this.props.children), height: 17}
   timerID: SharedTimerID
 
   constructor(props: Props) {
     super(props)
     // create a ref to store the textInput DOM element
-    this.boxRef = React.createRef()
+    this._boxRef
```
Comment (chrisnojima) [inline shared/chat/conversation/messages/wrapper/exploding-height-retainer/index.desktop.js]: entirely remove constructor

File: shared/chat/conversation/messages/wrapper/exploding-height-retainer/index.desktop.js (hunk 1 above)
Comment (zanderz) [reply shared/chat/conversation/messages/wrapper/exploding-height-retainer/index.desktop.js]: ok

File: shared/chat/conversation/messages/wrapper/exploding-height-retainer/index.desktop.js (hunk 3 above)
Comment (zanderz) [reply shared/chat/conversation/messages/wrapper/exploding-height-retainer/index.desktop.js]: ok

File: shared/chat/conversation/messages/wrapper/exploding-height-retainer/index.desktop.js (hunk 2 above)
Comment (zanderz) [reply shared/chat/conversation/messages/wrapper/exploding-height-retainer/index.desktop.js]: ok

File: shared/chat/conversation/messages/wrapper/exploding-height-retainer/index.desktop.js [hunk 4]
```diff
@@ -24,6 +23,8 @@ type State = {
   height: number,
 }
 class ExplodingHeightRetainer extends React.PureComponent<Props, State> {
+  _boxRef = React.createRef()
+  setHeight: () => void
```
Comment (chrisnojima) [inline shared/chat/conversation/messages/wrapper/exploding-height-retainer/index.desktop.js]: why is this here?

Comment (zanderz) [reply shared/chat/conversation/messages/wrapper/exploding-height-retainer/index.desktop.js]: I don't recognize that, no idea. Will nuke.
